vllm/entrypoints/utils.py

```diff
@@ -20,10 +20,53 @@
                                               CompletionRequest)
 from vllm.logger import init_logger
 from vllm.platforms import current_platform
+from vllm.transformers_utils.configs.speculators.base import (
+    SpeculatorsConfig, SpeculatorsProcessingError,
+    load_speculators_config_dict)
 from vllm.utils import FlexibleArgumentParser
 
 logger = init_logger(__name__)
 
+
+def detect_and_merge_speculators_args(
+        args: argparse.Namespace) -> argparse.Namespace:
+    """Detect speculators models and merge with CLI args.
+
+    CLI args take precedence.
+    """
+    # Early exits
+    if (not hasattr(args, 'model') or not args.model
+            or getattr(args, 'speculative_config', None)):
+        return args
+
+    config_dict = load_speculators_config_dict(args.model)
+    if config_dict is None:
+        return args
+
+    try:
+        vllm_config = SpeculatorsConfig.convert_speculators_to_vllm(
```

Comment:
Wont this be incorrect?
You're missing the update step specific to each algorithm https://github.com/neuralmagic/vllm/blob/0faf3cc3e84a83bac56492d297f2c9909fec7de9/vllm/transformers_utils/configs/speculators/base.py#L39?